Automattic/newspack-newsletters
# feat: preview html

### All Submissions:






### Changes proposed in this Pull Request:

1205078412187623-as-1205078412187635

Implements HTML preview before sending the newsletter:

### With provider

<img width="1325" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/e2cb0ff3-d4fa-4b2f-96b7-83dce264e9f8">

### With "Manual" provider

<img width="1294" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/50cef1ca-3218-4da7-bf14-6bfe56553ba5">

The editor should also be able to change the ad settings or send a test email in this modal screen.

A new "Preview Email" button also renders the HTML in a modal, but without the sending functionality:

<img width="495" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/a04b429e-6d89-4dc7-a0dc-f4fd3bb0d585">

<img width="1285" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/bda52797-768f-4f2d-b816-aea80e356fef">

### How to test the changes in this Pull Request:

1. Make sure you have active newsletters ads configured
2. Set your newsletter provider as "Manual" and draft a new newsletter
3. Click "Preview email" and confirm the modal renders with injected ads
4. Close the modal and click to Send
5. Toggle the "Disable automatic insertion of ads" on and confirm it updates the HTML
6. Back to Newsletters settings, change your provider to a supported ESP
7. Repeat steps 3-5
8. Send a test newsletter and confirm you receive the expected content
9. Send the newsletter and confirm you receive the expected content

### Other information:
